qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -221,83 +218,12 @@ def jobs(
-                               'infoQueue.status': 'PENDING_IN_QUEUE'}
-            elif status == JobStatus.CANCELLED:
-                this_filter = {'status': ApiJobStatus.CANCELLED.value}
-            elif status == JobStatus.DONE:
-                this_filter = {'status': ApiJobStatus.COMPLETED.value}
-            elif status == JobStatus.ERROR:
-                this_filter = {'status': {'regexp': '^ERROR'}}
-            else:
-                raise IBMQBackendValueError('unrecognized value for "status" keyword '
-                                            'in job filter')
-            api_filter.update(this_filter)
-
-        if job_name:
-            api_filter['name'] = job_name
-
-        if db_filter:
-            # status takes precedence over db_filter for same keys
-            api_filter = {**db_filter, **api_filter}
-
-        # Retrieve the requested number of jobs, using pagination. The API
-        # might limit the number of jobs per request.
-        job_responses = []
-        current_page_limit = limit
-
-        while True:
-            job_page = self._api.list_jobs_statuses(limit=current_page_limit,
-                                                    skip=skip,
-                                                    extra_filter=api_filter)
-            job_responses += job_page
-            skip = skip + len(job_page)
-
-            if not job_page:
-                # Stop if there are no more jobs returned by the API.
-                break
-
-            if limit:
-                if len(job_responses) >= limit:
-                    # Stop if we have reached the limit.
-                    break
-                current_page_limit = limit - len(job_responses)
-            else:
-                current_page_limit = 0
-
-        job_list = []
-        for job_info in job_responses:
-            kwargs = {}
-            try:
-                job_kind = ApiJobKind(job_info.get('kind', None))
-            except ValueError:
-                # Discard pre-qobj jobs.
-                break
-
-            kwargs['use_websockets'] = True
-            if self._credentials.proxies:
-                # Disable using websockets through proxies.
-                kwargs['use_websockets'] = False
-            if job_kind == ApiJobKind.QOBJECT_STORAGE:
-                kwargs['use_object_storage'] = True
-
-            job = IBMQJob(self, job_info.get('id'), self._api,
-                          creation_date=job_info.get('creationDate'),
-                          api_status=job_info.get('status'),
-                          **kwargs)
-            job_list.append(job)
-
-        return job_list
+        warnings.warn('backend.jobs() is deprecated and will be removed after '
+                      '0.5. Please use provider.backends.jobs(backend_name='
+                      '"{}") instead.'.format(self.name()), DeprecationWarning)
+
+        return self._provider.backends.jobs(
```

Comment:
If the intent of this change is just to allow job retrieval without a backend, then `provider.jobs()` seem to make more sense than `provider.backends.jobs()`.

Reply:
Unfortunately `provider.jobs()` is explicitly not defined in the proposal, right?

Reply:
There will be more than one job type going forward depending on the service. This is why this is the way it is at service level